### ISSUE TITLE
`pos_vp_state_machine_test` proptest failure

### ISSUE THREAD

**Issue by james-chf** [anoma/namada#431]
Caught this proptest failure in a CI run - https://github.com/anoma/namada/runs/8159509648?check_suite_focus=true - for `native_vp::pos::tests::pos_vp_state_machine_test`

For commit 1cf052c313d5ba3a8f070e3389f143b98789e530

The test does pass locally, and a rerun of the CI job passed

CI run logs attached: [logs_1605.zip](https://github.com/anoma/namada/files/9479546/logs_1605.zip)

**Comment by Fraccaman** [comment]
@tzemanovic

**Comment by tzemanovic** [comment]
thanks for report! I think this is the same issue as https://github.com/anoma/namada/issues/164, that should be fixed by #254. 

Btw, with proptest, you can take the seed from the failed run to reproduce the issue locally, just below the last stack trace at line 12924:
```
proptest: Aborting shrinking after the PROPTEST_MAX_SHRINK_ITERS environment variable or ProptestConfig.max_shrink_iters iterations (set 20 to a large(r) value to shrink more; current configuration: 20 iterations)
proptest: Saving this and future failures in /home/runner/work/namada/namada/tests/proptest-regressions/native_vp/pos.txt
proptest: If this test was run on a CI system, you may wish to add the following line to your copy of the file. (You may need to create it.)
cc e371ced4f2fb4f83bbb150ef05310f453c2c9a2554888d27ee82935fa9e7b65e
```
